src/pages/[...slug].astro

```diff
@@ -0,0 +1,31 @@
+---
+import { PAGES } from '@/constants/pages';
+import Layout from '@layouts/Layout.astro';
+import PageHeader from '@components/PageHeader.astro';
+
+export function getStaticPaths() {
+  return PAGES.map(({ slug, text }) => ({
+    params: { slug },
+    props: { text },
+  }));
+}
+
+const { slug } = Astro.params;
+const { text } = Astro.props;
+
+let Component;
+try {
+  console.log(`Trying to import component: ${slug}.tsx`);
+  const module = await import(`@/components/pages/${slug}.tsx`);
+  console.log({ module });
+  Component = module.default;
+} catch (error) {
+  console.error('Error importing component:', error);
+  return Astro.redirect('/404');
+}
```

Comment:
This is generally where I'm stuck- I can't get these dynamic imports working, though astro docs lead me to believe this should be fair game!!

Reply:
have also tried this same setup with `[slug].astro` as the path here